[WebGPU Swift] Fix crashing regression tests

tadeuzagallo opened the pull request:
#### 9ee3842efc07b32297d302c6039c7c42f494816f
<pre>
[WebGPU Swift] Fix crashing regression tests
<a href="https://bugs.webkit.org/show_bug.cgi?id=300510">https://bugs.webkit.org/show_bug.cgi?id=300510</a>
<a href="https://rdar.apple.com/162374042">rdar://162374042</a>

Reviewed by Mike Wyrzykowski and Geoffrey Garen.

There were 8 crashing tests under fast/webgpu/nocrash caused by 3 separate small
bugs in CommandEncoder.swift:
- A call to Range using `(offset, size)` instead of `(start, end)`
- One remaining instance of `m_lastErrorString` being implicitly unwrapped when it
  was nil
- There was a bit of a confusion with the types used in `timestampWriteIndex`,
  so I cleaned it up, but I believe the real issue was the definition of
  `WGPU_MTLCounterDontSample` not matching `MTLCounterDontSample`. I&apos;m not sure
  why the latter wasn&apos;t used originally, but it seems to work now.

* Source/WebGPU/WebGPU/CommandEncoder.swift:
(WebGPU.finish(_:)):
(WebGPU.timestampWriteIndex(_:)):
(WebGPU.beginRenderPass(_:)):
(WebGPU.beginComputePass(_:)):

Canonical link: <a href="https://commits.webkit.org/301396@main">https://commits.webkit.org/301396@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/fa4942821946fdbc7692e83887d9c4e7a27a3ae6

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/125644 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/45306 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/36056 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/132505 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/77528 "Built successfully") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/aa80fdf5-83e2-40b1-a4ac-7b68f913fde0) 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/45993 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/53868 "Built successfully") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/95717 "Failed to checkout and rebase branch from PR 52124") | [💥 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/63841 "An unexpected error occured. Recent messages:Printed configuration") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/fd55902f-06a5-4cc5-ae4a-5cc819d04a4a) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/128592 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/36764 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/112355 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/76212 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/3adce2b1-04c2-4618-80a2-ef537ed13310) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/35665 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/75977 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/106542 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/30754 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/135177 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/52439 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/40202 "Passed tests") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/104184 "15 flakes 56 failures") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/52885 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/108566 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/103913 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/26503 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/49272 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/27581 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/49658 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/52334 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/58136 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/51682 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/55035 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/53378 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

geoffreygaren commented on Source/WebGPU/WebGPU/CommandEncoder.swift:
```diff
@@ -160,7 +159,7 @@ extension WebGPU.CommandEncoder {
         result.ptr().setBufferMapCount(m_bufferMapCount)
         if m_makeSubmitInvalid {
-            result.ptr().makeInvalid(m_lastErrorString as String)
+            result.ptr().makeInvalid(m_lastErrorString != nil ? m_lastErrorString! as String : String("Invalid CommandEncoder"))
```
I believe you can say `m_lastErrorString as? String ?? "Invalid CommandEncoder"`

tadeuzagallo committed:
[WebGPU Swift] Fix crashing regression tests
https://bugs.webkit.org/show_bug.cgi?id=300510
rdar://162374042

Reviewed by Mike Wyrzykowski and Geoffrey Garen.

There were 8 crashing tests under fast/webgpu/nocrash caused by 3 separate small
bugs in CommandEncoder.swift:
- A call to Range using `(offset, size)` instead of `(start, end)`
- One remaining instance of `m_lastErrorString` being implicitly unwrapped when it
  was nil
- There was a bit of a confusion with the types used in `timestampWriteIndex`,
  so I cleaned it up, but I believe the real issue was the definition of
  `WGPU_MTLCounterDontSample` not matching `MTLCounterDontSample`. I'm not sure
  why the latter wasn't used originally, but it seems to work now.

* Source/WebGPU/WebGPU/CommandEncoder.swift:
(WebGPU.finish(_:)):
(WebGPU.timestampWriteIndex(_:)):
(WebGPU.beginRenderPass(_:)):
(WebGPU.beginComputePass(_:)):

Canonical link: https://commits.webkit.org/301396@main
